packages/devtools_app/lib/src/history_viewport.dart

```diff
@@ -59,17 +63,40 @@ class HistoryViewport<T> extends StatelessWidget {
       theme,
       child: Row(
         children: [
-          ToolbarAction(
-            icon: Icons.chevron_left,
-            onPressed: history.hasPrevious ? history.moveBack : null,
-          ),
-          ToolbarAction(
-            icon: Icons.chevron_right,
-            onPressed: history.hasNext ? history.moveForward : null,
-          ),
-          const SizedBox(width: denseSpacing),
-          const VerticalDivider(thickness: 1.0),
-          const SizedBox(width: defaultSpacing),
+          if (historyEnabled) ...[
+            ToolbarAction(
+              icon: Icons.chevron_left,
+              onPressed: history.hasPrevious
+                  ? () {
+                      history.moveBack();
+                      if (onChange != null) {
+                        onChange(
+                          history.current.value,
+                          history.peekNext(),
+                        );
+                      }
+                    }
+                  : null,
+            ),
+            ToolbarAction(
+              icon: Icons.chevron_right,
+              onPressed: history.hasNext
+                  ? () {
+                      final current = history.current.value;
+                      history.moveForward();
+                      if (onChange != null) {
+                        onChange(
+                          history.current.value,
+                          current,
```

Comment:
why not implement a `peekBack` method to mirror what is done above with `moveBack` and `peekNext`?